README.md

```diff
@@ -4,6 +4,103 @@
 [![GitHub release](https://img.shields.io/github/release/jenkinsci/kubernetes-credentials-provider-plugin.svg?label=release)](https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/releases/latest)
 [![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/kubernetes-credentials-provider.svg?color=blue)](https://plugins.jenkins.io/kubernetes-credentials-provider)
 
-Jenkins credentials provider for Kubernetes
+Complete documentation on GitHub page is [here](https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/)
 
-Documentation is [here](https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/)
+The *Kubernetes Credentials Provider* is a [Jenkins](https://jenkins.io) plugin to enable the retreival of [Credentials](https://plugins.jenkins.io/credentials) directly from Kubernetes.
+
+The plugin supports most common credential types and defines an [`extension point`](https://jenkins.io/doc/developer/extensions/kubernetes-credentials-provider/) that can be implemented by other plugins to add support for custom Credential types. 
+
+# Using
```

Comment:
do you think we need all of this information here to satisy the plugins.jenkins.io  page.

I think a couple of sentences explaingin what it does (not how to use it) would be enough for the plugin site (where I people would just want to get an overview of what the thing does, when looking for a solution to some problem?).

Reply:
Many (most?) plugins have a complete description like that, and no external more complete README.

I don't think it's an issue, on the contrary having more info on the plugin description page can help users to decide or not using a particular plugin.

Also, I've tried to keep this README the most close to the one in docs so any future updates would be simplified.

Reply:
the problem is not the amount of information here, but that there is now duplicate information to maintain/update, or let get completely out of date :)

Reply:
Agree. Let me try to resume this README so it won't be too much of a burden.

Reply:
Shortened in https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/pull/81/commits/3dd62fc5529fdce9a40d64224e9fac33a24fe892, WDYT?